### ISSUE TITLE
issues logging to stdout?

### ISSUE THREAD

**Issue by bshaffer** [GoogleCloudPlatform/php-docker#40]
Logging to **STDOUT** is not showing up in the log files. When on an instance itself, I see none of the [expected logs](https://github.com/GoogleCloudPlatform/php-docker/blob/master/php-nginx/supervisord.conf#L39) under `/var/log/app_engine`, but they are instead under `/var/log/app_engine/app`:

![screen shot 2016-01-06 at 2 56 43 pm](https://cloud.githubusercontent.com/assets/103941/12157318/deae1988-b485-11e5-8a9e-b929976a2c3b.png)

If I run `sudo docker logs [CONTAINER]`, I see the same logs as I see in the Cloud Logging UI, but none of my application logs show up here.

**Comment by tmatsuo** [comment]
`/var/log/app_engine` directory should be there within the container.

re: STDOUT
Isn't it connected to the web server output?

**Comment by bshaffer** [comment]
OK, I think I've found the problem. Everything `STDOUT-***` and `STDERR-***` is empty. However, `/var/log/app_engine/php-error.log` (in the container) contains exactly what I'd expect - logs from stdout.

On the instance (which is running the containers), this is (for some reason) in `/var/log/app_engine/app/php-error.log`. So my current theory is the Cloud Logger UI does not know where to find these logs.

**Comment by JustinBeckwith** [comment]
@dlorenc will the fluentd logger in the sidecar find log files in a subdirectory?  Ideally - stdout/stderr from PHP would show up as stdout/stderr logs in cloud logging.  Then supervisord would have a supervisord.log instead of taking over console.

**Comment by bshaffer** [comment]
note: [php.ini](https://github.com/GoogleCloudPlatform/php-docker/blob/80f3c2aa42ea3782121ab2e3d7194c576c7964d9/php-nginx/php.ini#L589)

**Comment by cedricziel** [comment]
Sidenote: The current [logrotate configuration](https://github.com/GoogleCloudPlatform/php-docker/blob/master/php-nginx/logrotate.app_engine) doesn't rotate logs in subdirectories.

**Comment by JustinBeckwith** [comment]
Also adding @andrewsg for his thoughts here on the fluentd question.

**Comment by andrewsg** [comment]
Sorry about the discrepancy between `/var/log/app_engine/` and `.../app/`. One is inside the container and one is outside; we intend to clean that up so it's consistent soon. That shouldn't be the root cause of this issue.

The issues simply seems to be that the filename `php-error.log` doesn't match any of our path patterns. Try changing the name to "app-php.log" (app*.log is the pattern). You can see the actual fluentd configuration here: https://github.com/GoogleCloudPlatform/appengine-sidecars-docker/blob/master/fluentd_logger/managed_vms.conf

**Comment by tmatsuo** [comment]
@cedricziel The logrotate configuration is correct. We don't have any subdirectories within the container.

**Comment by tmatsuo** [comment]
@bshaffer 

For me, I don't see anything in `/var/log/app_engine/php-error.log`. How do you emit the logs?

**Comment by bshaffer** [comment]
I'm using `error_log`, and also silex's built-in error handler is using it for 404s, 500s, etc.

**Comment by tmatsuo** [comment]
Ok, so let's change the log file names. #41

**Comment by tmatsuo** [comment]
@andrewsg Thanks. We'll change the filename of the error logs so that `app` category log shows our logs.

BTW I have a question. In the `app` category logs (app.access.log), there are bunch of requests like:
```
16:05:51.000 - - 08/Jan/2016:00:05:51 +0000 "GET /index.php" 404
```

I think these are for health checking requests (for some reason, the URL is not /_ah/health, but I assume it's after rewritten).

Do you know who is writing these entry in `app.access.log`? We don't define this log file name in our image.

**Comment by andrewsg** [comment]
No, I don't know where those are coming from. Is `app.access.log` the name of the logging stream visible in the Cloud Console UI, or is it the name of a filename? If it's a filename I'd use lsof (you might have to `apt-get update; apt-get install lsof` inside the container) to find out what is writing to it.

**Comment by tmatsuo** [comment]
@andrewsg Thanks :) It turns out that our config file has a variable to make that filename.

**Comment by tmatsuo** [comment]
Fixed by #42
